pkg/network/protocols/http/ebpf_gotls.go

```diff
@@ -266,21 +289,32 @@ func (p *GoTLSProgram) handleProcessStart(pid pid) {
 		return
 	}
 
-	hookedBin, present := p.hookedBinaries[stat.Ino]
-	if !present {
+	hookedBin := p.getHookedBinary(stat.Ino)
+	if hookedBin == nil {
 		hookedBin, err = p.hookNewBinary(binPath, stat.Ino)
 		if err != nil {
-			log.Debugf("could not hook new binary: %s", err)
+			log.Debugf("could not hook new binary (%d): %s", pid, err)
 			return
 		}
-		p.hookedBinaries[stat.Ino] = hookedBin
+		hookedBin.mTime = stat.Mtim
+		p.setHookedBinary(stat.Ino, hookedBin)
+	}
+
+	if stat.Mtim != hookedBin.mTime {
+		log.Warnf("binary %q has been modified since it has been hooked, skipping process registration.", binPath)
+		return
 	}
```

Comment:
```suggestion
	} else if stat.Mtim != hookedBin.mTime {
		log.Warnf("binary %q has been modified since it has been hooked, skipping process registration.", binPath)
		return
	}
```